managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1592,10 +1592,21 @@ synchronized void ledgerClosed(final LedgerHandle lh) {
     }
 
     synchronized void createLedgerAfterClosed() {
-        STATE_UPDATER.set(this, State.CreatingLedger);
-        this.lastLedgerCreationInitiationTimestamp = System.currentTimeMillis();
-        mbean.startDataLedgerCreateOp();
-        asyncCreateLedger(bookKeeper, config, digestType, this, Collections.emptyMap());
+        if(isNeededCreateNewLedgerAfterCloseLedger()) {
+            log.info("[{}] Creating a new ledger after closed", name);
+            STATE_UPDATER.set(this, State.CreatingLedger);
+            this.lastLedgerCreationInitiationTimestamp = System.currentTimeMillis();
+            mbean.startDataLedgerCreateOp();
+            asyncCreateLedger(bookKeeper, config, digestType, this, Collections.emptyMap());
+        }
+    }
+
+    synchronized boolean isNeededCreateNewLedgerAfterCloseLedger() {
```

Comment:
It seems that there is no need to add `synchronized` here
The following methods have `synchronized`, `isNeededCreateNewLedgerAfterCloseLedger()` is called inside them:
```
createLedgerAfterClosed()
ledgerClosed(final LedgerHandle lh)
internalAsyncAddEntry(OpAddEntry addOperation)
```

Reply:
yes, now it is not neccessary  to add synchronized before isNeededCreateNewLedgerAfterCloseLedger method. and add synchronized now avoid it is used by others.

Reply:
We don't know the logic of subsequent calls to this method, whether the same lock (ML object) is used.

Reply:
ok, i change it later.